File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
        ListNode* resList=new ListNode();
        ListNode* head=resList;
        
        while(l1 && l2){
            if(l1->val<l2->val){
                resList->next=l1;
                l1=l1->next;
            }    
            else{
                resList->next=l2;
                l2=l2->next;
            }
            
            resList=resList->next;
        }
        
        while(l1){
            resList->next=l1;
            l1=l1->next;
            resList=resList->next;
        }

        while(l2){
            resList->next=l2;
            l2=l2->next;
            resList=resList->next;
        }

        return head->next;
    }
};

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode fl=new ListNode();
        ListNode tmp=fl;
        
        while(l1!=null && l2!=null){
            if(l1.val<l2.val){
                tmp.next=l1;
                l1=l1.next;
            }
            else{
                tmp.next=l2;
                l2=l2.next;
            }
            tmp=tmp.next;
        }
       
        while(l1!=null){
            tmp.next=l1;
            tmp=tmp.next;
            l1=l1.next;
        }
        
        while(l2!=null){
            tmp.next=l2;
            tmp=tmp.next;
            l2=l2.next;
        }
        
        return fl.next;
    }
}